### DISCUSSION TITLE
Verbindung zu Vitosoft

### DISCUSSION THREAD

**Discussion by Phil1pp** [philippoo66/optolink-splitter#26] [General]
Kann ich den Splitter auch verwenden um über LAN/WLAN mit der Vitosoft auf die Heizung zuzugreifen?
Zumindest [hier ](https://github.com/sarnau/InsideViessmannVitosoft) scheint jemand schon mal etwas ähnlich gemacht zu haben.
Könnte man das zusammenführen?

**Comment by philippoo66** [comment]
moin @Phil1pp !

Vitosoft ist ein (Windows) Programm von Viessmann, eigentlich zu Servicezwecken und zur Inbetriebnahme. Der gute Makus 'sarnau' Fritze hat das nur 'auseinander genommen'. Die Optolink Kommunikationsbibliothek vom Splitter beruht auf seiner Arbeit. Vitosoft ist 3 Monate frei, danach kostet es richtig Geld. Das schmeisst dir auch dein ganzes System durcheinander mit eigenem Web Server, SQL Server und sonst noch was. Ist ein echtes Monster, am besten auf einem eigenen Compi oder in einer virtuellen Umgebung / auf einer virtuellen Maschine installieren. Vitosoft braucht eine direkte serielle Optolink Verbindung, ist also nicht netzwerkfähig. Man _könnte_ eine Bridge schreiben, die die serielle Kommunikation auf TCP/IP umsetzt (der Splitter kann das per raw verarbeiten), aber ich sehe da wenig Nutzen drin, es wäre aber beispiesweise unter Verwendung von com0com auch kein großer Aufwand.

Ich habe ja [Viessdata](https://github.com/philippoo66/ViessData21/wiki) etwas weiterentwickelt und das funktioniert (in der neuen Version) natürlich mit dem Splitter (nutzt da TCP/IP, also W/LAN). Das ist allerdings überwiegend für die Vitodens Gasthermen, im 'Übersichts-Tab' sind viele Adressen hard coded (das war schon immer so und ich hab mir nie die Mühe gemacht, das zu ändern, ist auch nicht ganz trivial wenn es wirklich dynamisch/'universell' sein soll). Die Datentabelle im 2. Tab geht aber mit jedem Optolink Gerät und auch die Grafiken, wenn man das entsprechend in den xmls konfiguriert.

Der Splitter dient ja nur als 'Weiche', um Datenströme zu lenken und zu 'übersetzen', er ist keine Anwendung im Sinne eines front-ends. Geläufige front-ends sind heutzutage diese Hausautomatisierungen wie ioBroker, Node Red, Home Assistat usw., die kannst du ja nach deinen Vorstellungen und Bedürfnissen gestalten. Die gehen dann über MQTT (unnd das über W/LAN). 

Du kannst mittels dem Splitter auch ganz 'pur' mit einem Terminal-Emulationsprogramm wie PuTTY Daten der Heizung lesen und schreiben, eben per TCP/IP.

Das läuft alles über W/LAN.

Grüsse!
Phil

**Comment by FransOv** [comment]
It is quite possible to use Vitossoft with Otpolink-Spliiter. Just unplug the usb cable from the Vitoconnect and plug it in your Windows computer. You might have to install a driver in Windows for the serialto USB device used on the RPI.
When you have the Vitosoft program it is quite useful to query the Vitosoft database directly with the help of Microsoft SQl Server Management Studio. You can either query the tables directly by device type or you can query based on the information Vitosoft has gathered from your system.
To query based on device type determine your device type from the table ecnDatapointType the code is in dp.Address. The query I find most usefull is:
```
select dp.Address as Model, 
et.Address as Parameter, 
substring(et.address,charindex('~',et.address)+1,6) Address, 
et.Conversion, 
case et.Type when 1 then 'R' when 2 then 'R'when 3 then'R/W' else '' end ReadWrite,
ete.ByteLength, et.DefaultValue, 
ev.Unit, ev.Stepping, ev.DataType, ev.ValuePrecision, ev.LowerBorder, ev.UpperBorder, 
string_AGG(ev.EnumAddressValue,',') as 'Values', string_AGG(ev.Description,',') as Descriptions
from ecnDeviceTypeDataPointTypeLink dev 
join ecnDatapointType dp on dev.DataPointTypeId=dp.Id 
join ecnDataPointTypeEventTypeLink dpe on dp.Id=dpe.DataPointTypeId
join ecnEventType et on et.id=dpe.EventTypeId
left outer join ecnEventTypeEventValueTypeLink etvl on etvl.EventTypeId=et.Id
left outer join ecnEventValueType ev on etvl.EventValueId=ev.Id
left outer join [vsmEventTypeExtension] ete on ete.EventTypeId=et.Id
where dp.Address='XXXXXXX'
group by et.Address, dp.Address,
 substring(et.address,charindex('~',et.address)+1,6),
 et.Conversion, case et.Type when 1 then 'R' when 2 then 'R'when 3 then'R/W' else '' end,
 ete.ByteLength, et.DefaultValue, ev.Unit, ev.Stepping, ev.DataType, ev.ValuePrecision, ev.LowerBorder, ev.UpperBorder
```
Where XXXXX is the value of ecnDatapointType .Address for your system.

If you have already connected Vitosoft to your system, have given it enough time to gather all the information abouf your system and have saved this information in the Vitosoft database, You can find the DeviceId of your system in ecnDevice and use this query:
```
Select dev.Name, et.Address, 
substring(et.address,charindex('~',et.address)+1,6) Parameter, 
et.Conversion, ete.ByteLength,
case et.Type when 1 then 'R' when 2 then 'R'when 3 then'R/W' else '' end ReadWrite,
ev.Unit, ev.Stepping, ev.DataType, ev.ValuePrecision, ev.LowerBorder, ev.UpperBorder, 
ev.DataType, ev.ValuePrecision, ev.LowerBorder, ev.UpperBorder, 
evc.ValueType, evc.Value_Binary, evc.Value_Bit, evc.Value_DateTime, 
evc.Value_Float, evc.Value_Int, evc.Value_Ntext, 
string_AGG(ev.EnumAddressValue,',') as 'Values', string_AGG(ev.Description,',') as Descriptions
from ecnDevice dev
 join ecnEventGroupValueCache evc on dev.Id=evc.DeviceId
 join ecnEventType et on evc.EventTypeId=et.Id
 join ecnEventValueType evt on evc.EventValueTypeId=evt.Id
 left outer join ecnEventTypeEventValueTypeLink etvl on etvl.EventTypeId=et.Id
 left outer join ecnEventValueType ev on etvl.EventValueId=ev.Id
 left outer join [vsmEventTypeExtension] ete on ete.EventTypeId=et.Id
where DeviceId=8
group by dev.Name, et.Address,
 substring(et.address,charindex('~',et.address)+1,6),
 et.Conversion, case et.Type when 1 then 'R' when 2 then 'R'when 3 then'R/W' else '' end,
 ete.ByteLength, et.DefaultValue, ev.Unit, ev.Stepping, ev.DataType, ev.ValuePrecision, ev.LowerBorder, ev.UpperBorder,
 evc.ValueType, evc.Value_Binary, evc.Value_Bit, evc.Value_DateTime, 
 evc.Value_Float, evc.Value_Int, evc.Value_Ntext, ev.DataType, ev.ValuePrecision, ev.LowerBorder, ev.UpperBorder 
 Order by Parameter
```
For DeviceId enter the Id of your system in ecnDevice.

**Be aware that there are several errors in the Database. I found several related to th Natural Cooling (NC) function!**
